index.js

```diff
@@ -140,6 +140,14 @@ class Matrix {
     negate() {
         return this.scale(-1);
     }
+    trace() {
+        var ret = 0;
+        for (var i = 0; i < this.data.length; i++) {
+            ret += this.data[i][i];
+        }
+
+        return ret;
+    }
```

Comment:
If we create a separate function to extract the diagonal as a vector, then this would simply be the manhattan norm of that vector. Do you think it would be more elegant to rewrite it like that?